src/plugins/intel_npu/src/al/include/intel_npu/config/compiler.hpp

```
@@ -379,4 +379,25 @@ struct COMPILER_DYNAMIC_QUANTIZATION final : OptionBase<COMPILER_DYNAMIC_QUANTIZ
     }
 };
 
+//
+// BATCH_COMPILER_MODE_SETTINGS
+//
+
+struct BATCH_COMPILER_MODE_SETTINGS final : OptionBase<BATCH_COMPILER_MODE_SETTINGS, std::string> {
```

What is happening when this property is set by the user? Is BATCH_MODE working automatically as COMPILER? Should we throw an error in case BATCH_MODE is set to PLUGIN and BATCH_COMPILER_MODE_SETTINGS has a valid value? (Maybe from compiler?)

---

good question,
I'd better throw an error from the compiler side, as a huge part of batch logic already implemented there